src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -0,0 +1,105 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group BGE
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Utilities to support Batch Gift Entry.
+*/
+public with sharing class BGE_BatchGiftEntry_UTIL {
+
+    /*******************************************************************************************************
+    * @description Checks for field permissions to create a new Batch/Template
+    * @param sObjectName string of the sObjectName
+    * @return void; throws an AuraHandledException if any fields fail FLS check
+    */
+    public static void checkFieldPermissions() {
+
+        List<DescribeFieldResult> fieldsToCheck = new List<DescribeFieldResult>();
+
+        List<String> fieldNames = getBatchFieldNames();
+
+        for (String field : fieldNames) {
+            if (field != 'Id') {
+                fieldsToCheck.add(UTIL_Describe.getFieldDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImportBatch__c'), UTIL_Namespace.StrTokenNSPrefix(field)));
+            }
+        }
+
+        // Donation Amount is also required
+        fieldsToCheck.add(UTIL_Describe.getFieldDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), UTIL_Namespace.StrTokenNSPrefix('Donation_Amount__c')));
+
+        for (DescribeFieldResult dfr : fieldsToCheck) {
+            if (!canUpdateField(dfr)) {
+                throw new AuraHandledException(Label.bgeFLSError);
+            }
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description checks for read, create, and edit FLS for a given field
+    * @param dfr DescribeFieldResult of the field to check
+    * @return Boolean
+    */
+    public static Boolean canUpdateField(DescribeFieldResult dfr) {
+        if (!dfr.isCreateable() || !dfr.isUpdateable()) {
```

Comment:
You could just write this as `return dfr.isCreateable() && dfr.isUpdateable()` #takeitorleaveit